### PR TITLE
GH-1946 Marshaller Serialize Requests

### PR DESCRIPTION
**Pull request by GordonSmith**
Add method to serializeRequests, which in turn can be used with primeData.

Fixes GH-1946

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jchambers-ln Please Review

**Comment by jchambers-ln** [comment]
Looks good.

File: src/marshaller/HipieDDL.js [hunk 1]
```diff
@@ -1287,7 +1319,7 @@
         var datasourceRequestOptimizer = new DatasourceRequestOptimizer();
         for (var key in this.visualizationRequests) {
             var item = this.visualizationRequests[key];
-            if (item.updateVisualization.type !== "GRAPH") {
+            if (!this.skipClear && item.updateVisualization.type !== "GRAPH") {
```
Comment (hamiltwp) [inline src/marshaller/HipieDDL.js]: primeData never clears the visualization but later VisualizationRequestOptimizer() may clear.  Just checking that this is intended.

Comment (GordonSmith) [inline src/marshaller/HipieDDL.js]: PrimeData will should be clearing the visualization here:  https://github.com/GordonSmith/Visualization/blob/c48e2463ad5d0cde256af0c79e6d6f16e18ec873/src/marshaller/HipieDDL.js#L1553-L1565

**Pull request merged**